modbusApp/src/modbusInterpose.c

```diff
@@ -343,7 +345,7 @@ static asynStatus readIt(void *ppvt, asynUser *pasynUser,
                 if (status != asynSuccess) return status;
                 if (nbytesActual >= 2) {
                     int id = ((pPvt->buffer[0] & 0xFF)<<8)|(pPvt->buffer[1]&0xFF);
-                    if (id == pPvt->transactionId) break;
+                    if ((pPvt->skipTransactionId != 0) && id == pPvt->transactionId) break;
```

Comment:
`(pPvt->skipTransactionId != 0 || id == pPvt->transactionId)` ?

Reply:
i think we want to do the stuff below if skipping transaction id is true don't we - this will break and not do anything after?

Reply:
i think latest commit is correct - if checking transaction id AND id is wrong then break,

Reply:
I think `if ((pPvt->skipTransactionId == 0) && id == pPvt->transactionId) break;` is still not quite correct, you want to break if skip or id is correct, above will never exit `for` except on error. The original code change change was https://github.com/epics-modules/modbus/commit/eaa3c1d27cb53c6cfebd7fe71b3d4ec8f0667af2

I think  `if (pPvt->skipTransactionId != 0 || id == pPvt->transactionId) break;` restores original logic, other than it will now want at least 2 bytes in a reply, which i think is fine.

Reply:
ah yes sorry, the break is for that for loop as well isn't it, i was being stupid